pkg/binlog/position_test.go

```diff
@@ -347,3 +349,196 @@ func (t *testPositionSuite) TestComparePosition(c *C) {
 		c.Assert(cmp, Equals, cs.cmp)
 	}
 }
+
+func (t *testPositionSuite) TestCompareCompareLocation(c *C) {
+	testCases := []struct {
+		flavor string
+		pos1   gmysql.Position
+		gset1  string
+		pos2   gmysql.Position
+		gset2  string
+		cmp    int
+	}{
+		{
+			// pos1 = pos2
+			gmysql.MySQLFlavor,
+			gmysql.Position{
+				Name: "binlog.00001",
+				Pos:  123,
+			},
+			"",
+			gmysql.Position{
+				Name: "binlog.00001",
+				Pos:  123,
+			},
+			"",
+			0,
+		}, {
+			// pos1 = pos2
+			gmysql.MariaDBFlavor,
+			gmysql.Position{
+				Name: "binlog.00001",
+				Pos:  123,
+			},
+			"",
+			gmysql.Position{
+				Name: "binlog.00001",
+				Pos:  123,
+			},
+			"",
+			0,
+		}, {
+			// pos1 < pos2
+			gmysql.MariaDBFlavor,
+			gmysql.Position{
+				Name: "binlog.00001",
+				Pos:  123,
+			},
+			"",
+			gmysql.Position{
+				Name: "binlog.00002",
+				Pos:  122,
+			},
+			"",
+			-1,
+		}, {
+			// pos1 > pos2
+			gmysql.MySQLFlavor,
+			gmysql.Position{
+				Name: "binlog.00003",
+				Pos:  123,
+			},
+			"",
+			gmysql.Position{
+				Name: "binlog.00002",
+				Pos:  122,
+			},
+			"",
+			1,
+		}, {
+			// gset1 = gset2, pos1 < pos2
+			gmysql.MySQLFlavor,
+			gmysql.Position{
+				Name: "binlog.00001",
+				Pos:  123,
+			},
+			"53ea0ed1-9bf8-11e6-8bea-64006a897c73:1-4",
+			gmysql.Position{
+				Name: "binlog.00002",
+				Pos:  122,
+			},
+			"53ea0ed1-9bf8-11e6-8bea-64006a897c73:1-4",
+			-1,
+		}, {
+			// compare by gtid set, gset2 contains gset1
+			gmysql.MySQLFlavor,
+			gmysql.Position{
+				Name: "binlog.00001",
+				Pos:  123,
+			},
+			"53ea0ed1-9bf8-11e6-8bea-64006a897c73:1-2,53ea0ed1-9bf8-11e6-8bea-64006a897c74:1-2",
+			gmysql.Position{
```

Comment:
Is location2.Pos < location1.Pos better?

Reply:
maybe mysql master switch to slave, the slave's binlog position is less than master.